hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/schema/SchemaVersionsRepositoryFactory.java

```diff
@@ -47,6 +44,8 @@ private ExecutorService getVersionsReloader() {
 
     @Override
     public void dispose(SchemaVersionsRepository instance) {
-
+        if (instance instanceof CachedSchemaVersionsRepository) {
+            ((CachedSchemaVersionsRepository) instance).shutdown();
+        }
```

Comment:
this is very nasty, maybe we should add shutdown to SchemaVersionRepository instead?